harper-core/src/document.rs

```diff
@@ -458,27 +461,75 @@ impl Document {
         ]))
     }
 
-    /// Assumes that the first matched token is the canonical one to be condensed into.
-    /// Takes a callback that can be used to retroactively edit the canonical token afterwards.
-    fn condense_expr<F>(&mut self, expr: &impl Expr, edit: F)
-    where
-        F: Fn(&mut Token),
-    {
-        let matches = expr.iter_matches_in_doc(self).collect::<Vec<_>>();
+    fn condense_dotted_latin(&mut self) {
+        self.condense_expr(&Self::DOTTED_LATIN_EXPR.with(|v| v.clone()), |_| {})
+    }
 
-        let mut remove_indices = VecDeque::with_capacity(matches.len());
+    // Loan phrases such as en masse
 
-        for m in matches {
-            remove_indices.extend(m.start + 1..m.end);
-            self.tokens[m.start].span = self.tokens[m.into_iter()].span().unwrap();
-            edit(&mut self.tokens[m.start]);
-        }
+    thread_local! {
+        static LOAN_PHRASES_EXPR: Lrc<FirstMatchOf> = Document::uncached_loan_phrases_expr();
+    }
+
+    fn uncached_loan_phrases_expr() -> Lrc<FirstMatchOf> {
+        Lrc::new(FirstMatchOf::new(
+            [
+                "ad nauseam",
+                "alma mater",
+                // "avant-garde",
+                "Bézier curve",
+                "bona fide",
+                // "cul-de-sac",
+                "de facto",
+                "de jure",
+                "de minimis",
+                "déjà vu",
+                "deja vu",
+                "en masse",
+                // "foo bar baz",
```

Comment:
This is about meta syntaxic variables.

https://en.wikipedia.org/wiki/Metasyntactic_variable

Is there a rule about them ?

Same question for Loren Ipsum ?

https://en.wikipedia.org/wiki/Lorem_ipsum

I'm sorry I ask without checking.

Is there a need for meta syntaxic and Lorem Ipsum in this rule, or in another ?